src/mapgen.h

```diff
@@ -114,6 +114,7 @@ enum MapgenType {
 	MAPGEN_V5,
 	MAPGEN_V6,
 	MAPGEN_V7,
+	MAPGEN_CARPATHIAN,
```

Comment:
I changed the enum to this order:
`
enum MapgenType {
	MAPGEN_V5,
	MAPGEN_V6,
	MAPGEN_V7,
	MAPGEN_FLAT,
	MAPGEN_FRACTAL,
	MAPGEN_VALLEYS,
	MAPGEN_SINGLENODE,
	MAPGEN_INVALID,
	MAPGEN_CARPATHIAN,
};
`


but it compiles with error:

/Users/vlapsley/games/minetest/src/mapgen.cpp:93:1: error: 'registered_mapgens_is_wrong_size'
      declared as an array with a negative size
STATIC_ASSERT(
^~~~~~~~~~~~~~
/Users/vlapsley/games/minetest/src/util/basic_macros.h:50:36: note: expanded from macro
      'STATIC_ASSERT'
        UNUSED_ATTRIBUTE typedef char msg[!!(expr) * 2 - 1]
                                          ^~~~~~~~~~~~~~~~
1 error generated.